File: app/src/main/java/bogomolov/aa/fitrack/view/DrawerMotionLayout.java
package bogomolov.aa.fitrack.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.constraintlayout.motion.widget.MotionLayout;
import androidx.drawerlayout.widget.DrawerLayout;

public class DrawerMotionLayout extends MotionLayout implements DrawerLayout.DrawerListener{

    public DrawerMotionLayout(Context context) {
        super(context);
    }

    public DrawerMotionLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public DrawerMotionLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void onDrawerSlide(@NonNull View drawerView, float slideOffset) {
        setProgress(slideOffset);
    }

    @Override
    public void onDrawerOpened(@NonNull View drawerView) {

    }

    @Override
    public void onDrawerClosed(@NonNull View drawerView) {

    }

    @Override
    public void onDrawerStateChanged(int newState) {

    }

    @Override
    protected void onAttachedToWindow() {
        super.onAttachedToWindow();
        ((DrawerLayout)getParent()).addDrawerListener(this);
    }
}

File: app/src/main/java/bogomolov/aa/fitrack/features/shared/DrawerMotionLayout.kt
package bogomolov.aa.fitrack.features.shared

import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.constraintlayout.motion.widget.MotionLayout
import androidx.drawerlayout.widget.DrawerLayout

class DrawerMotionLayout : MotionLayout, DrawerLayout.DrawerListener {

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun onDrawerSlide(drawerView: View, slideOffset: Float) {
        progress = slideOffset
    }

    override fun onDrawerOpened(drawerView: View) {}

    override fun onDrawerClosed(drawerView: View) {}

    override fun onDrawerStateChanged(newState: Int) {}

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        (parent as DrawerLayout).addDrawerListener(this)
    }
}
